raiwidgets/raiwidgets/model_analysis_dashboard_input.py

```diff
@@ -0,0 +1,357 @@
+from .constants import SKLearn
+import pandas as pd
+import numpy as np
+from scipy.sparse import issparse
+from .interfaces import WidgetRequestResponseConstants,\
+    ModelAnalysisDashboardData, Dataset, ModelExplanationData,\
+    PrecomputedExplanations, FeatureImportance, EBMGlobalExplanation,\
+    ErrorAnalysisConfig
+from .explanation_constants import ExplanationDashboardInterface
+import traceback
+from erroranalysis._internal.error_analyzer import ModelAnalyzer
+from responsibleai._internal.constants import ErrorAnalysisManagerKeys
+
+
+class ModelAnalysisDashboardInput:
+    def __init__(
+            self,
+            analysis: ModelAnalysis):
+        """Initialize the Explanation Dashboard Input.
+
+        :param analysis:
+            An ModelAnalysis object that represents an explanation.
+        :type analysis: ModelAnalysis
+        """
+        self._analysis = analysis
+        model = analysis.model
+        self._is_classifier = model is not None\
+            and hasattr(model, SKLearn.PREDICT_PROBA) and \
+            model.predict_proba is not None
+        self._dataframeColumns = None
+        self.dashboard_input = ModelAnalysisDashboardData()
+        self.dashboard_input.dataset = self._get_dataset()
+        self._feature_length = len(self.dashboard_input.dataset.featureNames)
+        self._row_length = len(self.dashboard_input.dataset.features)
+        self.dashboard_input.modelExplanationData = [
+            self._get_interpret(i) for i in self._analysis.explainer.get()]
+        self.dashboard_input.errorAnalysisConfig = [
+            self._get_error_analysis(i)
+            for i in self._analysis.error_analysis.list()["reports"]]
+        x_test = analysis.test.drop(columns=[analysis.target_column])
+        y_test = analysis.test[analysis.target_column]
+        self._error_analyzer = ModelAnalyzer(model, x_test,
+                                             y_test,
+                                             x_test.columns.values.tolist(),
+                                             analysis.categorical_features)
+
+    def on_predict(self, data):
+        try:
+            if self._dataframeColumns is not None:
+                data = pd.DataFrame(data, columns=self._dataframeColumns)
+            if (self._is_classifier):
+                prediction = self._convert_to_list(
+                    self._analysis.model.predict_proba(data))
+            else:
+                prediction = self._convert_to_list(
+                    self._analysis.model.predict(data))
+            return {
+                WidgetRequestResponseConstants.data: prediction
+            }
+        except Exception as e:
+            print(e)
+            traceback.print_exc()
+            return {
+                WidgetRequestResponseConstants.error: "Model threw exception"
+                " while predicting...",
+                WidgetRequestResponseConstants.data: []
+            }
+
+    def debug_ml(self, data):
```

Comment:
I'm assuming most of this file is duplicated from error analysis, right? If so, can we get that functionality in a common super class instead of duplicating it?

Reply:
yes,
I didn't do that because we will remove error analysis eventually, didn't want to handle difference through extra parameter to shared super class, or accidentally break existing error anslysis.